test/regress/run_regression.py

```diff
@@ -205,6 +233,8 @@ def run_regression(unsat_cores, proofs, dump, use_skip_return_code, wrapper,
             command_lines.append(line[len(COMMAND_LINE):])
         elif line.startswith(REQUIRES):
             requires.append(line[len(REQUIRES):].strip())
+        elif line.startswith(LOGIC):
+            logic = line[len(LOGIC):-1].strip()
```

Comment:
I have a slight preference to deal with this the same as with the `status_regex`. I would define a `logic_regex` and then use it to extract the logic. This seems more consistent since setting the logic depends on the file format.